x-pack/plugins/apm/public/context/url_params_context/helpers.ts

```diff
@@ -30,24 +41,37 @@ export function getDateRange({
 }) {
   // If the previous state had the same range, just return that instead of calculating a new range.
   if (state.rangeFrom === rangeFrom && state.rangeTo === rangeTo) {
-    return { start: state.start, end: state.end };
+    return {
+      start: state.start,
+      end: state.end,
+      exactStart: state.exactStart,
+      exactEnd: state.exactEnd,
+    };
   }
-
   const start = getParsedDate(rangeFrom);
   const end = getParsedDate(rangeTo, { roundUp: true });
 
   // `getParsedDate` will return undefined for invalid or empty dates. We return
   // the previous state if either date is undefined.
   if (!start || !end) {
-    return { start: state.start, end: state.end };
+    return {
+      start: state.start,
+      exactStart: state.start,
+      end: state.end,
+      exactEnd: state.end,
+    };
   }
 
   // rounds down start to minute
   const roundedStart = moment(start).startOf('minute');
+  const exactStart = getExactDate(rangeFrom) || roundedStart;
+  const exactEnd = getExactDate(rangeTo) || end;
```

Comment:
Either:

```suggestion
  const exactStart = getExactDate(rangeFrom) || start;
  const exactEnd = getExactDate(rangeTo) || end;
```

Or better:

```suggestion
  const exactStart = getExactDate(rangeFrom);
  const exactEnd = getExactDate(rangeTo);
```